bus-mapping/src/evm/opcodes/mload.rs

```diff
@@ -38,12 +38,21 @@ impl Opcode for Mload {
             StackOp::new(RW::READ, stack_position, stack_value_read),
         );
 
-        //
-        // First mem read -> 32 MemoryOp generated.
-        //
+        // Read the memory
         let mut mem_read_addr: MemoryAddress = stack_value_read.try_into()?;
         let mem_read_value = next_steps[0].memory().read_word(mem_read_addr)?;
 
+        //
+        // First stack write
+        //
+        ctx.push_op(
```

Comment:
Why change this order, are these two implementation(32 MemoryOp read first or stack write first) different?

Reply:
@Brechtpd?

I will merge it once this is resolved.

Reply:
To make the circuit implementation easier to do both MLOAD/MSTORE implementation. `Stack - Memory - Stack` only really makes sense for MLOAD, for MSTORE `Stack - Stack - Memory` is how you would order them normally. Having the order be the same for both (and MSTORE8 in a future PR) makes the circuit easier because otherwise we would have to have extra expressions to set `gc` correctly depending on the opcode.

So a single order is easier, and ``Stack - Stack - Memory` seems nicer to me because the different data locations are grouped together, but that's of course not really important, the only important thing is that they are consistent so I just picked that one.

Reply:
Nice, thanks for explanation:)

Reply:
Thanks @Brechtpd!